step-plans/step-plans-base-artefacts/src/main/java/step/artefacts/handlers/CallFunctionHandler.java

```diff
@@ -210,10 +211,13 @@ protected void execute_(CallFunctionReportNode node, CallFunction testArtefact)
 					session.put(AbstractFunctionHandler.EXECUTION_CONTEXT_KEY, new ExecutionContextWrapper(context));
 					session.put(AbstractFunctionHandler.ARTEFACT_PATH, currentArtefactPath());
 				}
-				
-				node.setAgentUrl(token.getAgent().getAgentUrl());
+
+				String agentUrl = token.getAgent().getAgentUrl();
+				node.setAgentUrl(agentUrl);
 				node.setTokenId(token.getID());
-				
+
+				context.get(ExecutionAgentUrls.class).add(agentUrl);

```

Comment:
We're also adding local agentUrl here. Not sure if we shouldn't track only "remote" agents

Reply:
I'm not sure if that is the case. Do you have an example? Note that the URL is sometimes null, in which case it's simply ignored -- I think this is what the "local" agent is?

Reply:
Well, it's implicitly checked in the add method anyway (which ignores null values), but I can also make it explicit.